File: API/DTOs/DeviceDto.cs
using API.Models;
using System.ComponentModel.DataAnnotations;

namespace API.DTOs
{
    public class DeviceResponseDto
    {
        public int DeviceId { get; set; }
        public int UserId { get; set; }
        public string Name { get; set; } = string.Empty;
        public string? Model { get; set; }
        public DeviceStatus Status { get; set; }
        public string? FirmwareVersion { get; set; }
        public DateTime? LastSync { get; set; }
        public string? SerialNumber { get; set; }
        public DateTime CreatedAt { get; set; }
        public DateTime UpdatedAt { get; set; }
    }

    public class CreateDeviceDto
    {
        [Required]
        [MaxLength(100)]
        public string Name { get; set; } = string.Empty;

        [MaxLength(100)]
        public string? Model { get; set; }

        [MaxLength(100)]
        public string? SerialNumber { get; set; }
    }

    public class UpdateDeviceDto
    {
        [Required]
        [MaxLength(100)]
        public string Name { get; set; } = string.Empty;

        [MaxLength(100)]
        public string? Model { get; set; }

        public DeviceStatus Status { get; set; }

        [MaxLength(50)]
        public string? FirmwareVersion { get; set; }
    }
}

File: Mobile/composeApp/src/androidMain/kotlin/com/plantiq/data/model/DeviceDto.kt
package com.plantiq.data.model

import com.google.gson.annotations.SerializedName

data class DeviceResponseDto(
    @SerializedName("deviceId") val deviceId: Int,
    @SerializedName("userId") val userId: Int,
    @SerializedName("name") val name: String,
    @SerializedName("model") val model: String?,
    @SerializedName("serialNumber") val serialNumber: String?,
    @SerializedName("status") val status: String?
)

data class CreateDeviceDto(
    @SerializedName("name") val name: String,
    @SerializedName("model") val model: String?,
    @SerializedName("serialNumber") val serialNumber: String?
)
